### PR TITLE
feat: Add support for `REQUEST_MEMBERS` packet sending 

### PR DESCRIPTION
**Pull request by Damego**
## About

This pull request implements support for `REQUEST_MEMBERS` packet sending and `GUILD_MEMBERS_CHUNK` event receiving

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [x] To add a new feature
  - [x] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/api/gateway/client.py [hunk 1]
```diff
@@ -658,11 +670,23 @@ def __modify_guild_cache(
...
+                elif "_chunk" in name:
+                    _objs = getattr(obj, attr, None)
+                    if _objs:
+                        for _obj in _objs:
+                            for index, __obj in enumerate(iterable):
+                                if __obj.id == _obj.id:
+                                    iterable[index] = _obj
+                                    break
+                            else:
+                                iterable.append(_obj)
```
Comment (Toricane) [inline interactions/api/gateway/client.py]:
```suggestion
            elif ids is not None:
                objs = getattr(obj, attr, None)
                if "_update" in name:
                    if objs is not None:
                        iterable.clear()
                        iterable.extend(objs)
                elif "_chunk" in name:
                    if objs is not None:
                        for _obj in objs:
                            for index, __obj in enumerate(iterable):
                                if __obj.id == _obj.id:
                                    iterable[index] = _obj
                                    break
                            else:
                                iterable.append(_obj)
```

**Review by Toricane — CHANGES_REQUESTED** [review]
Small refactor

File: interactions/api/gateway/client.py [hunk 2]
```diff
@@ -658,11 +670,22 @@ def __modify_guild_cache(
...
+                        iterable.extend(objs)
+                elif "_chunk" in name:
+                    if objs is not None:
+                        for _obj in objs:
+                            for index, __obj in enumerate(iterable):
+                                if __obj.id == _obj.id:
+                                    iterable[index] = _obj
+                                    break
+                            else:
+                                iterable.append(_obj)
```
Comment (Toricane) [inline interactions/api/gateway/client.py]:
```suggestion
                if objs is not None:
                    if "_update" in name:
                        iterable.clear()
                        iterable.extend(objs)
                    elif "_chunk" in name:
                        for _obj in objs:
                            for index, __obj in enumerate(iterable):
                                if __obj.id == _obj.id:
                                    iterable[index] = _obj
                                    break
                            else:
                                iterable.append(_obj)
```

**Review by Toricane — CHANGES_REQUESTED** [review]
One more thing

File: interactions/api/gateway/client.py (hunk 2 above)
Comment (Damego) [reply interactions/api/gateway/client.py]: ~~Can you use walrus operator to get ‘objs‘ in if statement above?~~
I will do it

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**